src/include/duckdb/main/extension_entries.hpp

```diff
@@ -509,7 +509,7 @@ static constexpr ExtensionEntry EXTENSION_FILE_CONTAINS[] = {{".parquet?", "parq
 // TODO: automate by passing though to script via duckdb
 static constexpr ExtensionEntry EXTENSION_SECRET_TYPES[] = {
     {"s3", "httpfs"},   {"r2", "httpfs"},          {"gcs", "httpfs"},
-    {"azure", "azure"}, {"huggingface", "httpfs"}, {"bearer", "httpfs"}}; // EXTENSION_SECRET_TYPES
+    {"azure", "azure"}, {"huggingface", "httpfs"}, {"http", "httpfs"}}; // EXTENSION_SECRET_TYPES
```

Comment:
I guess this needs adding a `{"http", "duckdb"}`, and handling the mapping `""` -> already in duckdb core lib (or some other convention).

Reply:
Thanks for the pointers. I updated the python script as well. Re: @samansmink 's comment, wonder if it's okay to just leave the extension field empty (so no ext is autoloaded for the http secret type). And when `bearer` secret type becomes user-facing, it can be re-added to the list? Currently "huggingface" is also going to try to autoload httpfs extension so that should cover bearer anyways?

Reply:
I've updated `scripts/generate_extensions_function.py` but somehow CI is still failing? Any idea how to fix?

Reply:
[src/include/duckdb/main/extension_entries.hpp](https://github.com/duckdb/duckdb/pull/14496/files/d4fea3163e49bae97c15d0782bbc98d85143f84e#diff-0efec0604cc77f847f57cdc8c3f16f0bfa12b0899a7e2aa305142cf4aed4fd64) maps out functions, types etc that live in extension so that duckdb can autoload the correct extension or throw an error hinting the user which extension to load to make it available.

Maybe I'm misunderstanding something, but the HTTP secret types are [always loaded](https://github.com/duckdb/duckdb/blob/dfdd7968262d912910d8249bde3524e068c67713/src/main/secret/secret_manager.cpp#L61) so should not be in this list at all?

Reply:
The motivation is that we need a way to keep track of newly supported DDB secret types (we need to do some sync logic between client and server) and we’ve been using this list as an indication (but http was missing and bearer isn’t user facing yet). Wonder if there’s a different way to get this info?

Reply:
Perhaps we can add a metadata function that lists available secret types? e.g. `FROM duckdb_secret_types()` @samansmink thoughts?

Reply:
sure, sounds good to me!

Reply:
Thanks! Closing thie PR in favor of https://github.com/duckdblabs/motherduck/issues/187